test/TensorFlowRuntime/runtime_entry_points.swift

```diff
@@ -0,0 +1,28 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+// REQUIRES: swift_test_mode_optimize
+
+import CTensorFlow
+import TensorFlow
+import TensorFlowUnittest
+import StdlibUnittest
+
+var RuntimeEntryPointTests = TestSuite("RuntimeEntryPoint")
+
+RuntimeEntryPointTests.testCPUOrGPU("RoundTrip_CTensorHandle_AnyTensorHandle") {
+  let zero: TensorHandle<Float> =
+    #tfop("Const", dtype: Float.self, value$tensor: 0.0)
+  var cHandle = _GetCTensorHandleFromSwift(zero as _AnyTensorHandle)
+  let status = TF_NewStatus()
+  // We must do a copy, i.e. a retain on the tensor handle, to make sure it won't
+  // get double-free'd when both `zero` and `anyHandle` below go out of scope.
+  cHandle = TFE_TensorHandleCopySharingTensor(cHandle, status)
```

Comment:
hmm this suggests using `TensorHandle<...>(owning: cHandle)` within `_GetCTensorHandleFromSwift()` is not correct. A c handle should not be **owned** by 2 or more TensorHandle objects.

In IRgen, for a given tensor handle `x`, we may need to call `_GetCTensorHandleFromSwift()` on it multiple times, if `x` is consumed by multiple tfops. We don't want IRGen code to call `TFE_TensorHandleCopySharingTensor` to rebalance the ref count.

Can we call TFE_TensorHandleCopySharingTensor() within `_GetCTensorHandleFromSwift()`?

Reply:
As we discussed in person, making `_GetCTensorHandleFromSwift` _not_ make a copy is the correct behavior since the lifetime of each `graph_op` argument is guaranteed by construction.

This test case is irrelevant because it's making a new `TensorHandle` own the extracted `CTensorHandle`, which is for testing purpose and will not be emitted by IRGen.